cmd/homi/setup/flags.go

```diff
@@ -29,382 +29,422 @@ var (
-		Name:  "validators-num",
-		Usage: "Number of validators. If not set, it will be set the number of option cn-num",
-		Value: 0,
+		Name:    "validators-num",
+		Usage:   "Number of validators. If not set, it will be set the number of option cn-num",
+		Aliases: []string{"topology.validators-num"},
+		Value:   0,
 	}
-
 	numOfPNsFlag = cli.IntFlag{
-		Name:  "pn-num",
-		Usage: "Number of proxy node",
-		Value: 0,
+		Name:    "pn-num",
+		Usage:   "Number of proxy node",
+		Aliases: []string{"topology.pn-num"},
+		Value:   0,
 	}
-
 	numOfENsFlag = cli.IntFlag{
-		Name:  "en-num",
-		Usage: "Number of end-point node",
-		Value: 0,
+		Name:    "en-num",
+		Usage:   "Number of end-point node",
+		Aliases: []string{"topology.en-num"},
+		Value:   0,
 	}
-
 	numOfSCNsFlag = cli.IntFlag{
-		Name:  "scn-num",
-		Usage: "Number of service chain nodes",
-		Value: 0,
+		Name:    "scn-num",
+		Usage:   "Number of service chain nodes",
+		Aliases: []string{"topology.scn-num"},
+		Value:   0,
 	}
-
 	numOfSPNsFlag = cli.IntFlag{
-		Name:  "spn-num",
-		Usage: "Number of service chain proxy nodes",
-		Value: 0,
+		Name:    "spn-num",
+		Usage:   "Number of service chain proxy nodes",
+		Aliases: []string{"topology.spn-num"},
+		Value:   0,
 	}
-
 	numOfSENsFlag = cli.IntFlag{
-		Name:  "sen-num",
-		Usage: "Number of service chain end-point nodes",
-		Value: 0,
+		Name:    "sen-num",
+		Usage:   "Number of service chain end-point nodes",
+		Aliases: []string{"topology.sen-num"},
+		Value:   0,
 	}
-
 	numOfTestKeyFlag = cli.IntFlag{
-		Name:  "test-num",
-		Usage: "Number of test key",
-		Value: 0,
+		Name:    "test-num",
+		Usage:   "Number of test key",
+		Aliases: []string{"topology.test-num"},
+		Value:   0,
 	}
 
+	genesisTypeFlag = cli.StringFlag{
+		Name:    "genesis-type",
+		Usage:   "Set the type of genesis.json to generate (cypress-test, cypress, baobab-test, baobab, clique, servicechain, servicechain-test, istanbul)",
+		Aliases: []string{"genesis.type"},
+	}
 	chainIDFlag = cli.Uint64Flag{
-		Name:  "chainID",
-		Usage: "ChainID",
-		Value: 1000,
+		Name:    "chainID",
+		Usage:   "ChainID",
+		Aliases: []string{"genesis.chain-id"},
+		Value:   1000,
 	}
 
 	serviceChainIDFlag = cli.Uint64Flag{
-		Name:  "serviceChainID",
-		Usage: "Service Chain ID",
-		Value: 1001,
+		Name:    "serviceChainID",
+		Usage:   "Service Chain ID",
+		Aliases: []string{"genesis.service-chain-id"},
+		Value:   1001,
 	}
 
 	unitPriceFlag = cli.Uint64Flag{
-		Name:  "unitPrice",
-		Usage: "Price of unit",
-		Value: 0,
+		Name:    "unitPrice",
+		Usage:   "Price of unit",
+		Aliases: []string{"genesis.unit-price"},
+		Value:   0,
 	}
 
 	deriveShaImplFlag = cli.IntFlag{
-		Name:  "deriveShaImpl",
-		Usage: "Implementation of DeriveSha [0:Original, 1:Simple, 2:Concat]",
-		Value: 0,
+		Name:    "deriveShaImpl",
+		Usage:   "Implementation of DeriveSha [0:Original, 1:Simple, 2:Concat]",
+		Aliases: []string{"genesis.derive-sha-implement"},
```

Comment:
```suggestion
		Aliases: []string{"genesis.derive-sha-impl"},
```
either `impl` or `implementation` seems appropriate.